RD-6890 Extended login error messages mapping mechanism

Vorbert-Kruk opened the pull request:
<!--- Provide JIRA issue ID and a general summary of your changes in the Title above -->
<!--- e.g. "RD-1234 Improve styling in SuperComponent" -->
<!--- Note that the title is used as a commit message after merging the PR -->

## Description
<!--- Describe your changes to help reviewers understand the details. -->

## Screenshots / Videos
<!--- If applicable, add screenshot(s) or video(s) describing the changes you made. -->
<!--- Consider adding comparison screenshot(s) - before and after the change. -->
<!--- If not applicable, just write “N/A” in this section. -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [ ] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [ ] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [ ] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [ ] I added proper labels to this PR.

## Tests
<!--- If applicable, describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- If applicable, provide a link to system tests, e.g: -->
<!--- [#1039](https://jenkins.cloudify.co/.../1039) (2021-08-06 07:05:50) --> 
<!--- If not applicable, just write “N/A” in this section. -->

## Documentation
<!--- If applicable, describe how you documented or plan to document your changes. -->
<!--- Check UI documentation guidelines - https://cloudifysource.atlassian.net/l/c/EYWJ0XfB - for details. -->
<!--- Provide links to JIRA issues, other PRs or related documentation pages. -->
<!--- If not applicable, just write “N/A” in this section. -->

qooban reviewed (dismissed):
Hmm... I'm approving not to block it, but I suspect it's a bit overcomplicated. Probably we could avoid validation state variables and should stop using login error from Redux store in `LoginPage` component.

I did a quick try to simplify it - not fully tested, just to share general idea - https://github.com/cloudify-cosmo/cloudify-stage/pull/2451
Could you let me know what do you think? 

Alternatively you can use derived state from props (and pass `loginError` to some kind of state variable only when `loginError` changes), but AFAIR that's not recommended by React - https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops.

Vorbert-Kruk commented:
> Hmm... I'm approving not to block it, but I suspect it's a bit overcomplicated. Probably we could avoid validation state variables and should stop using login error from Redux store in `LoginPage` component.
> 
> I did a quick try to simplify it - not fully tested, just to share general idea - #2451 Could you let me know what do you think?
> 
> Alternatively you can use derived state from props (and pass `loginError` to some kind of state variable only when `loginError` changes), but AFAIR that's not recommended by React - https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops.

@qooban I think that the code share by you have indeed a useful simplification, which we could use 🎉 
To be honest, I was a little bit shortsighted on the solution, because of the previous way of implementation 😄 
But we can totally go with your idea and stop using data from store in order of using the local state 💪 
Thanks a lot for spotting that direction and for providing initial implementation of it 💪 
I'll adjust the code in a similar way 🎉

Vorbert-Kruk commented:
@qooban I've applied mentioned simplification 💪 
Also I've adjusted and refactored it a little bit, to make the solution fully working 🎉 
Additionally, new round of system tests have been executed 🚀 

Besides that, I can see that `auth.error` redux field is not being used anywhere except in [ManagerStatePersister file](https://github.com/cloudify-cosmo/cloudify-stage/blob/master/app/utils/ManagerStatePersister.ts#L23)
Because of that, I'll probably create a ticket for removing this property, as it doesn't seem to bring any value 💪

Vorbert-Kruk commented:
Unit test were failing after adding `throw error` in `login` action
I've adjusted them within b03475621bb6fd6e0be47b034ac21fc884498e6f 🎉
